main.nf

```diff
@@ -517,9 +531,19 @@ if (!params.Q2imported){
 			discard_untrimmed = params.retain_untrimmed ? '' : '--discard-untrimmed'
 			"""
 			mkdir -p trimmed
-			cutadapt -g ${params.FW_primer} -G ${params.RV_primer} ${discard_untrimmed} \
-				-o trimmed/$folder${params.split}${reads[0]} -p trimmed/$folder${params.split}${reads[1]} \
-				${reads[0]} ${reads[1]} > cutadapt_log_${pair_id}.txt
+			if [[ $params.double_primer=TRUE && $discard_untrimmed="--discard-untrimmed" ]]; then
+				mkdir -p firstcutadapt
+				cutadapt -g ${params.FW_primer} -G ${params.RV_primer} ${discard_untrimmed} \
+					-o firstcutadapt/$folder${params.split}${reads[0]} -p firstcutadapt/$folder${params.split}${reads[1]} \
+					${reads[0]} ${reads[1]} >> cutadapt_log_${pair_id}.txt
+				cutadapt -g ${params.FW_primer} -G ${params.RV_primer} --discard-trimmed \
+                                        -o trimmed/$folder${params.split}${reads[0]} -p trimmed/$folder${params.split}${reads[1]} \
+                                        firstcutadapt/$folder${params.split}${reads[0]} firstcutadapt/$folder${params.split}${reads[1]} >> cutadapt_log_${pair_id}.txt
```

Comment:
[QIIME2](https://forum.qiime2.org/t/identical-asvs-with-and-without-adapter-from-cutadapt-dada2/17512/4) has an option `-p-twice` for cutadapt. I am not suggesting to use cutadapt in QIIME2, but probably QIIME2 is using a native parameter of cutadapt to remove double primers? If yes, it would make the task simpler.

Reply:
I haven't been able to find this option for cutadapt. There is an option to trim away more than one occurrence of the primer, but since we have only seen double primers as a result of bad PCR we want to discard the entire read instead of just trimming away the primer. Therefore I think this is the best approach, but there might be better options.

Reply:
Yes, I see your point regarding removing those reads. I am not aware of a better solution, just came across that info and thought to better share it. Alright, than this is fine with me.